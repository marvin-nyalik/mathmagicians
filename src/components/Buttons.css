.keys {
  font-family: Arial, Helvetica, sans-serif;
  width: 25%;
  height: 4rem;
  text-align: center;
  font-size: 24px;
  font-weight: 500;
}

.operator-keys {
  font-family: Arial, Helvetica, sans-serif;
  width: 25%;
  height: 4rem;
  text-align: center;
  font-size: 24px;
  font-weight: 500;
  background: orange;
}

.operator-keys:hover {
  background: lightsalmon;
}

.zero-key {
  font-family: Arial, Helvetica, sans-serif;
  width: 50%;
  height: 4rem;
  text-align: center;
  font-size: 24px;
  font-weight: 500;
}

