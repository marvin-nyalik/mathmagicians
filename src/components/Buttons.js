import PropTypes from 'prop-types';
import './Buttons.css';

const Buttons = ({ value, handleInput }) => {
  if (value === '+' || value === '=' || value === '-' || value === 'x' || value === '\u00F7') {
    return <button className="operator-keys" type="button" onClick={handleInput} key={value}>{value}</button>;
  }
  if (value === '0') {
    return <button className="zero-key" type="button" onClick={handleInput} key={value}>{value}</button>;
  }
  return <button className="keys" type="button" onClick={handleInput} key={value}>{value}</button>;
};

Buttons.propTypes = {
  value: PropTypes.string.isRequired,
  handleInput: PropTypes.func.isRequired,
};

export default Buttons;
