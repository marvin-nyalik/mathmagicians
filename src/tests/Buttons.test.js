import { render, fireEvent } from '@testing-library/react';
import Buttons from '../components/Buttons';

describe('Tests for the Button component', () => {
  const value = '+';
  const handleInput = jest.fn();

  it('The button matches the snapshot', () => {
    const { container } = render(<Buttons value={value} handleInput={handleInput} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the correct inner text', () => {
    const value = '9';
    const { getByText } = render(<Buttons value={value} handleInput={handleInput} />);
    const renderedBtn = getByText(value);
    expect(renderedBtn).toBeTruthy();
  });

  it('calls the handleInput function when clicked', () => {
    const { getByText } = render(<Buttons value="0" handleInput={handleInput} />);
    const button = getByText('0');
    fireEvent.click(button);
    expect(handleInput).toHaveBeenCalledTimes(1);
  });
});
