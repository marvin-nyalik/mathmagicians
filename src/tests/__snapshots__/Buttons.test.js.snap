// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the Button component The button matches the snapshot 1`] = `
<button
  class="operator-keys"
  type="button"
>
  +
</button>
`;
